assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -65,32 +105,49 @@ export const actions = {
 		};
 	},
 
-	receiveAccountsPropertiesProfilesCompleted() {
-		return {
-			payload: {},
-			type: RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_COMPLETED,
-		};
+	*resetAccounts() {
+		const registry = yield Data.commonActions.getRegistry();
+
+		yield { type: RESET_ACCOUNTS };
+
+		return registry.stores[ STORE_NAME ].getActions()
+			.invalidateResolutionForStoreSelector( 'getAccounts' );
 	},
 
-	receiveAccountsPropertiesProfilesFailed( error ) {
-		invariant( error, 'error is required.' );
+	*selectAccount( accountID ) {
+		invariant( isValidAccountSelection( accountID ), 'A valid accountID is required to select.' );
 
-		return {
-			payload: { error },
-			type: RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_FAILED,
-		};
+		const registry = yield Data.commonActions.getRegistry();
+		registry.dispatch( STORE_NAME ).setAccountID( accountID );
+		registry.dispatch( STORE_NAME ).setPropertyID( '' );
+		registry.dispatch( STORE_NAME ).setInternalWebPropertyID( '' );
+		registry.dispatch( STORE_NAME ).setProfileID( '' );
+
+		if ( ACCOUNT_CREATE === accountID ) {
+			return;
+		}
+
+		// Trigger cascading selections.
+		const properties = registry.select( STORE_NAME ).getProperties( accountID );
+		if ( properties === undefined ) {
+			return; // Selection will happen in resolver.
+		}
+		const property = properties[ 0 ] || { id: PROPERTY_CREATE };
+		registry.dispatch( STORE_NAME ).selectProperty( property.id );
```

Comment:
This would be different than the current behavior. With the current set up component, the property and profile preselect the first available option when you select a different account. Changing this would require additional changes to E2E tests as well, if not others.